### ISSUE TITLE
Password Recovery Mechanism

### ISSUE THREAD

**Issue by poikilotherm** [IQSS/dataverse#7290]
Per recent questions and discussions, I got the impression it might be useful to implement a password recovery mechanism for `dataverseAdmin` and API unblock-key.

My idea is easy:
Provide a config option (JVM option/MPCONFIG) to point to a file on disk, containing a new secret.

During deployment, if the file exists  and is readable by the appserver, the secrets would be updated in the database.

**Comment by donsizemore** [comment]
Assuming that the file should be ignored if accessible by any user beside the Dataverse service account, and that the jvm-option should remain disabled or removed during normal operation, I think this is a fine idea even outside of deployment.

**Comment by djbrooke** [comment]
Thanks @poikilotherm and thanks @donsizemore for the endorsement. 

@scolapasta @qqmyers any concerns?

**Comment by qqmyers** [comment]
A couple quick thoughts:

I agree that some way to regain admin rights is important. However, I note that there's a
POST /api/admin/authenticatedusers
and
POST /api/admin/superuser/{identifier}

that allow you to create a new user and assign them superadmin rights. The former is protected by the Builtin user key (a setting) and can be disabled.

There's also a delete user admin api call.

With these, is it already possible to delete the dataverseAdmin account, create a new one, and make it a superuser?

If not, given the power we already have in the admin api, would it be better to just consider a 'resetadminpassword' admin api call that is similarly restricted (perhaps requiring the builtin user key as well) rather than creating something that works differently?

**Comment by DonRichards** [comment]
I think this feature is worth reconsidering, particularly for administrative recovery and edge cases.

There are legitimate situations where a Dataverse administrator needs to reset the password of an existing builtin account without access to the email-based reset workflow. A shared service account used for API or cron jobs is an obvious example, especially when the associated mailbox is no longer accessible.

Administrators with sufficient privileges can already create users, manage roles, deactivate accounts, and perform other sensitive account operations. Password recovery for a builtin account does not seem fundamentally different. There should be a supported and auditable way to do it.

Without one, administrators are pushed toward worse options: temporarily changing the account email, modifying the database directly, or replacing the account entirely. Those approaches introduce more risk, not less.

The capability could be deliberately narrow. Restrict it to superusers. Require localhost or admin API access. Log the operation.

But there should be a recovery path.

Edge cases eventually become operational problems. A supported mechanism is better than leaving administrators to improvise one.

**Comment by qqmyers** [comment]
FWIW: Going through the password recovery page creates a recovery token in the db (passwordresetdata table). If the email can't be delivered, the token can still be used to reset the password (adding ?token={token} on the passwordreset.xhtml page). 

A superuser only way to view those tokens could make them more easily accessible. If that's of interest, I'd suggest a new issue as this one is primarily about the dataverseAdmin account itself (when you've lost access to the only superuser account), etc.

**Comment by pdurbin** [comment]
> There are legitimate situations where a Dataverse administrator needs to reset the password of an existing builtin account

I'm strongly reminded of this issue:

- #2029

It was closed with "we can't do this for security reasons" but @DonRichards if you want to create a fresh issue, please feel free! (Sounds like @qqmyers is also suggesting a separate issue.) It's been over a decade and I'm ok with revisiting this.

Also...

> A shared service account used for API or cron jobs

I'd feel remiss if I didn't mention that just this morning we talked about whether or not we should discourage shared accounts like this. There's some related discussion here: https://github.com/IQSS/dataverse-frontend/issues/1021#issuecomment-5333313323
